oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/ClusterNodeInfo.java

```diff
@@ -1067,10 +1067,9 @@ public boolean renewLease() throws DocumentStoreException {
                             "anymore. {}", id, doc);
                     // break here and let the next lease update attempt fail
                     break;
-                } else if (doc.getLeaseEndTime() == previousLeaseEndTime
-                        || doc.getLeaseEndTime() == updatedLeaseEndTime) {
-                    // set lease end times to current values
-                    previousLeaseEndTime = doc.getLeaseEndTime();
+                } else if (doc.getRuntimeId().equals(runtimeId)) {
```

Comment:
This may produce a NullPointerException. `doc.getRuntimeId()` may return `null`. Better flip it around.
```suggestion
                } else if (runtimeId.equals(doc.getRuntimeId())) {
```